packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -386,6 +386,72 @@ function useMediaUploadQueue() {
         isTranscoding: state.queue.some((item) => item.state === 'TRANSCODING'),
         isMuting: state.queue.some((item) => item.state === 'MUTING'),
         isTrimming: state.queue.some((item) => item.state === 'TRIMMING'),
+        isResourceProcessing: (resourceId) =>
+          state.queue.some(
+            (item) =>
+              [
+                'PENDING',
+                'UPLOADING',
+                'TRANSCODED',
+                'TRANSCODING',
+                'MUTING',
+                'MUTED',
+                'TRIMMING',
+                'TRIMMED',
+              ].includes(item.state) &&
+              item.resource.originalResourceId === resourceId
+          ),
+        isResourceProcessingById: (resourceId) =>
```

Comment:
Since we talked about improving naming of these functions (because the `ById` suffix is quite confusing), some ideas off the top of my head:

* Instead of `isResourceProcessingById`, use `isCurrentResourceProcessing`

(same would go for the other functions)

Reply:
Just wanted to ask about the naming here -- since `isResourceProcessing` and `isResourceProcessingById` both seem to be used by ID.

Reply:
https://github.com/google/web-stories-wp/pull/9785#discussion_r758331003

Reply:
@miina With the additional context, do you have any suggestions for improved naming here?

Reply:
Maybe `isResourceProcessingById` => `isOriginResourceProcessing`?

If `isResourceProcessing` is only used for the new resource, then maybe actually changing that to `isNewResourceProcessing` or `isUpdatedResourceProcessing` or `isGeneratedResourceProcessing` could also make things more clear. If it's used in some other cases as well then this wouldn't work.

Reply:
Fixed.